src/extensions/idle.rs

```diff
@@ -40,6 +65,21 @@ pub enum WaitOutcome {
     MailboxChanged,
 }
 
+/// Return type for IDLE response callbacks. Tells the IDLE connection
+/// if it should continue monitoring the connection or not.
+#[derive(Debug, PartialEq, Eq)]
+pub enum CallbackAction {
```

Comment:
I went back and forth on this a bit. I can go either way? The only reason I did this is because I find it easier to read and remember what the return value _means_ if it is a named value. I don't foresee any other behaviours, so this is purely in the service of being descriptive.

I feel like if I were coming back to callback code that I wrote in a year I would be looking at the `return true` or `return false` and then would have to look up what those meant, whereas `Stop` and `Continue` is fairly unambiguous?

Perhaps I am overthinking it? I have worked in codebases before with these exact same semantics (override this function and return true or false to indicate continue or not) and after some time we changed it to returning a named type because it was just more clear what was intended when coming back to modify code that hadn't been touched in awhile.

Reply:
I could go either way on this — the extra type does also increase the API surface, but perhaps in a small enough way that the improved clarity is worth it. I think maybe it'd help me if the name was more descriptive as well. How about:
```rust
enum OnResponse {
```

Reply:
I think I'll just go with `bool` here. You don't seem sold, and I don't feel strongly enough to haggle over names :)